pkgs/development/python-modules/filedate/default.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pip,
+  python-dateutil,
+}:
+buildPythonPackage rec {
+  pname = "filedate";
+  version = "3.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "kubinka0505";
+    repo = pname;
+    rev = version;
+    hash = "sha256-HvuGP+QlUlfAUfFmaVVvtPHGdrbWVxghQipnqTTvAQc=";
+  };
+
+  unpackPhase = ''
+    cp -r $src/Files/* .
+    cp -r $src/ReadMe.md .
+  '';
+
+  # The repo stores everything in "src" and uses setup.py to move "src" ->
+  # "filedate" before calling setup() and then tries to rename "filedate" back
+  # to "src" after.
+  postPatch = ''
+    mv src filedate
+    substituteInPlace setup.py \
+      --replace-fail "__title__ = os.path.basename(os.path.dirname(os.path.dirname(__file__)))" '__title__ = "filedate"'
+    substituteInPlace setup.py \
+      --replace-fail "cleanup = True" "cleanup = False"
+
+    # Read the ReadMe.md in our patched location
+    substituteInPlace setup.py \
+      --replace-fail 'desc = open("../ReadMe.md", encoding = "UTF-8").read().rstrip()' 'desc = open("ReadMe.md", encoding = "UTF-8").read().rstrip()'
+
+    # Disable renaming "filedate" dir back to "src"
+    substituteInPlace setup.py \
+      --replace-fail "if os.path.exists(__title__):" ""
+    substituteInPlace setup.py \
+      --replace-fail "	os.rename(__title__, directory)" ""
+  '';
+
+  buildInputs = [ pip ];
+
+  propagatedBuildInputs = [ python-dateutil ];
+
+  pythonImportsCheck = [ "filedate" ];
+
```

Comment:
Could you please add `pytestCheckHook` to `nativeCheckInputs` and enable tests?

Reply:
So I added this but got the error:
```
error: builder for '/nix/store/m79kf8n0n7lizncdfkc23a9h731rfhlp-python3.11-filedate-3.0.drv' failed with exit code 5;
       last 10 log lines:
       > Check whether the following modules can be imported: filedate
       > Running phase: pytestCheckPhase
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.11.9, pytest-8.1.1, pluggy-1.4.0
       > rootdir: /build
       > collected 0 items
       >
       > ============================ no tests ran in 0.00s =============================
       > /nix/store/558iw5j1bk7z6wrg8cp96q2rx03jqj1v-stdenv-linux/setup: line 1579: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/m79kf8n0n7lizncdfkc23a9h731rfhlp-python3.11-filedate-3.0.drv'.

```
How do I add `filedate` package as an input to it's own tests?

Reply:
Please add `pytestFlagsArray = [ "tests/unit.py" ];`.

see also https://docs.pytest.org/en/7.2.x/explanation/goodpractices.html#conventions-for-python-test-discovery
and https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

Reply:
I added this and `test_created` fails with a wrong datetime comparison so I've also added:

```
  pytestFlagsArray = [
    "tests/unit.py"
    "-k"
    # test_created fails with wrong datetime
    "\"not test_created\""
  ];
```

Reply:
Please follow the documentation and use `disableTests`.

Reply:
Oooohhh thank you for hand holding me through this. Updated.